src/librustc_middle/sir.rs

```diff
@@ -86,6 +86,8 @@ impl SirFuncCx<'tcx> {
                 flags |= ykpack::bodyflags::TRACE_HEAD;
             } else if attr.check_name(sym::trace_tail) {
                 flags |= ykpack::bodyflags::TRACE_TAIL;
+            } else if attr.check_name(sym::do_not_trace) {
```

Comment:
This mechanism has changed in upstream recently, so we will get a merge conflict at some point. Depending on if we merge the upstream sync before or after.

Reply:
Since the upstream sync was just merged, this'll need syncing.

Reply:
I didn't get a conflict here.

Reply:
Ah, but I did get an compile error.